### PR TITLE
Formula to find median in Binary Search changed

### PR DESCRIPTION
**Pull request by Souravjyoti**
### **Describe your change:**

Finding median using the previous formula may carry the median value out of range. Therefore, updated the formula with a better one.

### **Checklist:**

- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized.
- [ ] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ] All new Java files are placed inside an existing directory.
- [ ] All filenames are in all uppercase characters with no spaces or dashes.
- [ ] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Searches/BinarySearch.java [hunk 1]
```diff
@@ -45,7 +45,8 @@
     if (right < left) return -1; // this means that the key not found
 
     // find median
-    int median = (left + right) >>> 1;
+    //int median = (left + right) >>> 1;
+    int median = left + ((right-left) >>> 1)
```
Comment (shellhub) [inline Searches/BinarySearch.java:49]:
```suggestion
    int median = left + ((right-left) >>> 1)
```

File: Searches/BinarySearch.java [hunk 2]
```diff
@@ -45,7 +45,7 @@
     if (right < left) return -1; // this means that the key not found
 
     // find median
-    int median = (left + right) >>> 1;
+    int median = left + ((right-left) >>> 1)
```
Comment (shellhub) [inline Searches/BinarySearch.java:48]:
```suggestion
    int median = left + ((right-left) >>> 1);
```

**Review by shellhub — APPROVED** [review]
Thanks for your contribution.

File: Searches/BinarySearch.java [hunk 3]
```diff
@@ -45,7 +45,7 @@
     if (right < left) return -1; // this means that the key not found
 
     // find median
-    int median = (left + right) >>> 1;
+    int median = left + ((right-left) >>> 1);
```
Comment (nirvanarsc) [inline Searches/BinarySearch.java:48]: Actually this change is not necessary. Above expression is already using a [logical shift](https://en.wikipedia.org/wiki/Logical_shift) (three `>>>`) which will preserve the sign bit if a potential overflow happens when summing (left + right). For languages like C++ and others that only support [arithmetic shift](https://en.wikipedia.org/wiki/Arithmetic_shift) (two `>>`) above workaround is necessary.

TL;DR original code works fine
